webrender/src/device/gfx/device.rs

```diff
@@ -2076,6 +2260,25 @@ impl<B: hal::Backend> Device<B> {
         }
     }
 
+    pub fn copy_cache_buffer(&mut self, dst: &Texture, src: &Texture) {
+        let mut src_buffer = self.gpu_cache_buffers.remove(&src.id).unwrap();
+        {
+            let dst_buffer = self.gpu_cache_buffers.get_mut(&dst.id).unwrap();
+            let read_size = src_buffer.transit_range_end;
+            let (read_mapping, _) = src_buffer.map(&self.device, None);
+            unsafe {
+                let (write_mapping, _) = dst_buffer.map(&self.device, Some(read_size));
+                let src_slice = slice::from_raw_parts(read_mapping.ptr().as_ptr() as _, read_size as usize);
+                let dst_slice = slice::from_raw_parts_mut(write_mapping.ptr().as_ptr() as _, read_size as usize);
+                dst_slice.copy_from_slice(src_slice);
+            }
+            unsafe { dst_buffer.flush_mapped_ranges(&self.device, std::iter::once(0..read_size)) };
```

Comment:
isn't this done by Rendy-memory if we are using its mapping routines?

Reply:
I found using the `write` and `read` functions form `rendy-memory` here causes segfault (without any backtrace) after a certain time when running wrench tests.

Reply:
are you running wrench tests in Debug? please file a bug to rendy, it seems pretty important to investigate

Reply:
Created an issue : https://github.com/amethyst/rendy/issues/183

Reply:
It turned out that the issue was on our side see: https://github.com/amethyst/rendy/issues/183#issuecomment-519095649

Reply:
Removed this in the last commit: https://github.com/szeged/webrender/pull/297/commits/ed9e894391d366dfc69d34e7ef4b33250011e111#diff-cdf03c96bd34cb98c2d09b0be7658232L2275